core/gui/src/app/dashboard/service/user/download/download.service.ts

```diff
@@ -88,40 +88,70 @@ export class DownloadService {
     );
   }
 
+  /**
+   * Export the workflow result. If destination = "local", the server returns a BLOB (file).
+   * Otherwise, it returns JSON with a status message.
+   */
   public exportWorkflowResult(
     exportType: string,
     workflowId: number,
     workflowName: string,
-    operatorId: string,
-    operatorName: string,
+    operatorIds: string[],
     datasetIds: number[],
     rowIndex: number,
     columnIndex: number,
-    filename: string
+    filename: string,
+    destination: string // "local" or "dataset"
```

Comment:
for typescript, you can specify the type using the possible values. In this case, destination can have type: `destination: "local" || "dataset"`